### PR TITLE
LoopSimplify: strip dependency on DA (NFC)

### PR DESCRIPTION
**Pull request by artagnon**
Since no passes compute DependenceAnalysis via the PassManager, there is no value in preserving it here. Hence, strip the unnecessary dependency on DependenceAnalysis.

### CODE REVIEW

File: llvm/lib/Transforms/Utils/LoopSimplify.cpp [hunk 1]
```diff
@@ -756,13 +751,8 @@ namespace {
       AU.addRequired<LoopInfoWrapperPass>();
```
Comment (fhahn) [inline llvm/lib/Transforms/Utils/LoopSimplify.cpp:751]: is the legacy pass still used in tree?

**Review by nikic — CHANGES_REQUESTED** [review]
The legacy pass is still used by the backend, and at least some of these preservations are important.

File: llvm/lib/Transforms/Utils/LoopSimplify.cpp (hunk 1 above)
Comment (artagnon) [reply llvm/lib/Transforms/Utils/LoopSimplify.cpp:751]: Yes, unfortunately, as noticed by @nikic and the CI failure in test/CodeGen.

**Comment by artagnon** [comment]
> The legacy pass is still used by the backend, and at least some of these preservations are important.

The CI failures should be fixed now, but how can we be confident that the patch is correct? What is this mysterious preservation?

**Comment by artagnon** [comment]
Gentle ping. Is this patch okay now?

**Review by nikic — APPROVED** [review]
LGTM. It looks like passes that use DependenceAnalysis don't compute it via the pass manager anyway. It's possible that what LoopSimplify is doing could be useful in the future if we have passes using DA via the PM, but for now this looks fine to me.

**Pull request merged**